src/geofabrics/dem.py

```diff
@@ -1670,62 +1754,158 @@ def _create_data_set(
+            )
+        )
+        data_extents = geopandas.GeoDataFrame(
+            geometry=data_extents.clip(full_extents, keep_geom_type=True)
+        )
+        no_data_extents = full_extents.overlay(data_extents, how="difference")
+        # Keep areas without LiDAR data above the area threshold
+        no_data_extents = no_data_extents.explode(index_parts=False)
+        no_data_extents = no_data_extents[no_data_extents.area > area_threshold]
+
+        # Check if enough without LiDAR to use coarse DEMs
+        if len(no_data_extents) == 0:
+            logging.INFO(
+                f"Not on land areas greater than {area_threshold} "
+                "without LiDAR values. Ignoring all coarse DEMs."
+            )
+            return False
+
+        # Define a mask of the area without LiDAR data where there is now data
         z = self._dem.z.copy(deep=True)
         z.data[:] = 0
-        mask = numpy.logical_not(
-            numpy.isnan(z.rio.clip(region_to_rasterise.geometry, drop=False).data)
-        )
+        mask = z.rio.clip(no_data_extents.geometry, drop=False).notnull().data
+
+        # Iterate through DEMs
+        coarse_dem_added = False
+        logging.info(f"Incorporating coarse DEMs: {coarse_dem_paths}")
+        for coarse_dem_path in coarse_dem_paths:
+            logging.info(f"\tLoad coarse DEM: {coarse_dem_path}")
+            coarse_dem = CoarseDem(
+                dem_file=coarse_dem_path,
+                catchment_geometry=self.catchment_geometry,
+                set_foreshore=self.drop_offshore_lidar,
+                extent=no_data_extents,
+            )
+            # Add the coarse DEM data where there's no LiDAR updating the extents
+            if not coarse_dem.empty:
+                logging.info(f"\t\tAdd data from coarse DEM: {coarse_dem_path.name}")
+                if chunk_size is None:
+                    self.add_coarse_dem_no_chunking(
+                        coarse_dem=coarse_dem,
+                        region_to_rasterise=no_data_extents,
+                        mask=mask,
+                    )
+                else:
+                    self.add_coarse_dem_chunked(
+                        coarse_dem=coarse_dem,
+                        region_to_rasterise=no_data_extents,
+                        chunk_size=chunk_size,
+                        mask=mask,
+                    )
+                coarse_dem_added = True
+        # If coarse DEM data added update the data_source layer
+        if coarse_dem_added:
+            # Use a mask of the area without LiDAR data where there is now data
+            z = self._dem.z.copy(deep=True)
+            z.data[:] = 0
+            mask = z.rio.clip(no_data_extents.geometry, drop=False).notnull().data
+            self._dem["data_source"].data = numpy.where(
+                mask & self._dem.z.notnull().data,
+                self.SOURCE_CLASSIFICATION["coarse DEM"],
+                self._dem.data_source.data,
+            )
```

Comment:
I would recommend to use xarray.where here too :).

Reply:
cool - I've moved to do this in all cases - weirdly I seem to need to reenforce meta-data settings after - not quite sure why but only requires one reset of meta data so I've decided this is ok.

Reply:
@rosepearson oh, that would be a good reason for not using xarray... which metadata is lost? the rioxarray ones? or others?
also https://docs.xarray.dev/en/stable/generated/xarray.where.html mentions a `keep_attrs` parameter that could help?

Reply:
done thanks